app/server/utils.py

```diff
@@ -405,6 +425,26 @@ def paint(self, documents):
         return res
 
 
+class SpacyPainter(JSONPainter):
+
+    def paint(self, documents):
+        data = super().paint(documents)
+        res = []
+        for d in data:
+            text = d['text']
+            annotations = d.pop('annotations')
+            entities = []
+            for a in annotations:
+                label_id = a["label"]
+                label_text = Label.objects.get(id=label_id).text
```

Comment:
This will do a lot of calls to the database: number of data lines times number of annotations per data line.

Ideally we'd do a single trip to the database that fetches all the relevant label texts. Something along these lines perhaps?

```py
label_ids = {a['label'] for d in data for a in d['annotations']}
label_texts = {l['id']: l['text'] for l in Label.objects.filter(id__in=label_ids).values('text', 'id')}

for d in data:
  # ...
  for a in annotations:
    label_text = label_texts[a['label']]
```

(Sample code is just suggested as a starting point and wasn't tested or run.)